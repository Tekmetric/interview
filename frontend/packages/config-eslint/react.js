module.exports = {
  extends: [
    '@vercel/style-guide/eslint/browser',
    '@vercel/style-guide/eslint/typescript',
    '@vercel/style-guide/eslint/react'
  ].map(require.resolve),
  parserOptions: {
    project: '**/tsconfig.json',
    sourceType: 'module'
  },
  globals: {
    JSX: true
  },
  settings: {
    'import/resolver': {
      typescript: {
        project: '**/tsconfig.json',
        sourceType: 'module'
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx']
      }
    }
  },
  ignorePatterns: ['node_modules/', 'dist/', '.eslintrc.js', '**/*.css'],
  // add rules configurations here
  rules: {
    'import/no-default-export': 'off',
    'import/no-relative-packages': 'off',
    'react/button-has-type': 'off',
    'no-restricted-imports': [
      'error',
      {
        patterns: [
          {
            group: ['packages/*'],
            message: "Please use '@tekmetric/*' imports instead"
          }
        ]
      }
    ],
    'import/order': [
      'error',
      {
        'newlines-between': 'always',
        groups: [
          ['builtin', 'external'],
          'unknown',
          ['parent', 'sibling', 'index']
        ],
        pathGroups: [
          {
            pattern: '~/**',
            group: 'unknown',
            position: 'after'
          }
        ]
      }
    ],
    'react/function-component-definition': [
      'error',
      {
        namedComponents: 'arrow-function',
        unnamedComponents: 'arrow-function'
      }
    ]
  },
  overrides: [
    {
      files: ['*.config.js'],
      env: {
        node: true
      }
    },
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        'no-unused-vars': 'off',
        '@typescript-eslint/no-invalid-void-type': 'off',
        'react/jsx-no-leaked-render': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            vars: 'all',
            args: 'after-used',
            ignoreRestSiblings: true,
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_'
          }
        ],
        '@typescript-eslint/no-misused-promises': [
          'error',
          {
            checksVoidReturn: false
          }
        ]
      }
    },
    {
      files: ['*.test.ts', '*.test.tsx'],
      env: {
        jest: true
      },
      rules: {
        '@typescript-eslint/no-unsafe-call': 'off',
        '@typescript-eslint/no-unsafe-member-access': 'off'
      }
    }
  ]
}
