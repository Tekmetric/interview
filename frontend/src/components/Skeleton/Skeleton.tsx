import type { FC } from 'react';
import styles from './skeleton.module.css';

interface SkeletonProps {
  count: number;
}

const Skeleton: FC<SkeletonProps> = ({ count = 1 }) => {
  return (
    <div className={styles.skeleton}>
      {[...Array(count)].map((el, key) => (
        <div key={`skeleton_${key}`} />
      ))}
    </div>
  );
};

export default Skeleton;
