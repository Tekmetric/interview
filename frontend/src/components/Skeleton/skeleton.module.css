@keyframes skeleton-loading {
  0% {
    background-color: hsl(200, 20%, 80%);
  }
  100% {
    background-color: hsl(200, 20%, 95%);
  }
}

.skeleton {
  width: 100%;
  height: 100%;
  max-width: 100%;
}

.skeleton div {
  animation: skeleton-loading 1s linear infinite alternate;
  border-radius: var(--border-radius-md);
  height: 100%;
  opacity: 0.2;
}
