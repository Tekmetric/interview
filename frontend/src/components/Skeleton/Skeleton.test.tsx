import { render } from '@testing-library/react';
import Skeleton from './Skeleton';

describe('Skeleton', () => {
  it('renders correctly', () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });

  it('renders the correct number of blocks', () => {
    const { container } = render(<Skeleton count={3} />);
    expect(container.firstChild?.childNodes.length).toBe(3);
  });

  it('renders one block by default', () => {
    const { container } = render(<Skeleton />);
    expect(container.firstChild?.childNodes.length).toBe(1);
  });
});
