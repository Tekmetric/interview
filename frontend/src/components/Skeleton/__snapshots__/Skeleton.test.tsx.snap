// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > renders correctly 1`] = `
<div>
  <div
    class="_skeleton_bdf928"
  >
    <div />
  </div>
</div>
`;
